lib/Mojo/mysql/Native/Database.pm

```diff
@@ -0,0 +1,227 @@
+package Mojo::mysql::Native::Database;
+use Mojo::Base 'Mojo::mysql::Database';
+
+use Mojo::mysql::Connection;
+use Mojo::mysql::Native::Results;
+use Mojo::mysql::Native::Transaction;
+use Mojo::mysql::Util qw(expand_sql parse_url);
+use Scalar::Util 'weaken';
+use Carp 'croak';
+
+has 'connection';
+
+sub DESTROY {
+  my $self = shift;
+  return unless my $c = $self->connection;
+  return unless my $mysql = $self->mysql;
+  $mysql->_enqueue($c);
+}
+
+sub backlog { scalar @{shift->{waiting} || []} }
+
+sub begin {
+  my $self = shift;
+  $self->query('START TRANSACTION');
+  $self->query('SET autocommit=0');
```

Comment:
Could you rather do $self->connect->begin_work instead of those two query() calls? That way, the connection could also have state that would make begin() die if it's already in a transaction... Just like DBI.